### DISCUSSION TITLE
How to load data based on the uri in Splat / Catch-All Routes

### DISCUSSION THREAD

**Discussion by GuillaumeLaroucheUQAT** [TanStack/router#1017] [Q&A] [answered]
I have a splat route that looks like this : 

```
export const Route = new FileRoute("/services/$").createRoute({
	component: ServicePage,
	pendingComponent: ServicePageLoading,
	loader: ({ params }) => prefetchServicePage({ uri: `services/${params._splat}` })},
});
```
This code looks fine at first but I have 2 issues with this approach :

1. According to the doc, the params prop should have the type `{ _splat: string; }` but I have `{ "": string; }` so I need to add additional typecheck in the loader to make it works :
```
loader: ({ params }) => {
    const uri = "_splat" in params && typeof params._splat === "string" ? params._splat : "";
    return prefetchServicePage({ uri: `services/${uri}` });
},
```

2. Navigating to subroutes doesn't update the params prop in the loader so I get stale data and the component isn't rerendered even though I use an external library for data loading (SWR) and I have the prop `defaultPreloadStaleTime` set to 0 in my router. Setting other cache props didn't work.

Am I doing something wrong?

**Comment by schiller-manuel** [comment, marked as answer]
the typing is fixed in v1.5.8
can you please provide a minimal example on codesandbox for the second issue?

**Reply by GuillaumeLaroucheUQAT** [reply]
Thanks for the fix, that was fast! I'm unable to provide a minimal reproduction example at the moment but I will try to do it soon. However, if it helps, I have noticed in the Devtools that splat/catch-all routes uses the route path as the key for the cache while dynamic routes uses the pathname.

For exemple, a dynamic route `/services/$serviceId` with the param `service-a` will use the pathname `/services/service-a` as the key but the splat route `/services/$` will always use `/services/$`. This means the cache will always return the instance of the first match for a splat route.

I don't if it is intentionnal or not but I hope it can help!

**Reply by GuillaumeLaroucheUQAT** [reply]
@schiller-manuel Here is a minimal reproduction exemple : [https://stackblitz.com/edit/tanstack-router-rlxprr?file=src%2Froutes%2F%24.tsx](https://stackblitz.com/edit/tanstack-router-rlxprr?file=src%2Froutes%2F%24.tsx)

The exemple feature a splat page with a link that generate sub page. If the link is clicked, nothing happen unless the page is refreshed.

**Reply by schiller-manuel** [reply]
please create an bug report in an issue
